packages/customer/src/services/customer-group.ts

```diff
@@ -1,7 +1,7 @@
 import { DAL } from "@medusajs/types"
 import { ModulesSdkUtils } from "@medusajs/utils"
 import { CustomerGroup } from "@models"
-import { CreateCustomerGroupDTO, UpdateCustomerGroupDTO } from "@types"
+import { CreateCustomerGroupDTO, UpdateCustomerGroupDTO } from "@medusajs/types"
```

Comment:
suggestion: Ideally, internal services should use internal module types

Reply:
These are eventually used for `moduleService.createCustomerGroup/updateCustomerGroup` which are part of the public API of the module so thinking they are good to have available?

Reply:
The DTO from the public API and the internal services can differ at some point cause they have different responsibility in each layer. So yes we can reuse the one from the types package but it goes against the responsibility principle. The idea is that if the internal service DTO change then you can change them in the module types and not break the public API

Reply:
Being able to modify one without the other is great but would probably punt that concern until it's actually needed so we eliminate as much boilerplate for modules as possible.

Reply:
I see, well I follow whatever you choose guys, I gave my arguments and explanation ahah.

Reply:
It's not that it adds implementation time - it's that if you need to create 2-3 additional types for every service and you forget to add one or have a mistake in one of the types or similar, it sets an entire PR review cycle on hold for something that is not immediately useful.